libnopegl/src/block.c

```diff
@@ -197,15 +197,15 @@ int ngli_block_add_field(struct block *s, const char *name, int type, size_t cou
     return 0;
 }
 
-void ngli_block_field_copy(const struct block_field *fi, uint8_t * restrict dst, const uint8_t * restrict src)
+void ngli_block_field_copy_count(const struct block_field *fi, uint8_t * restrict dst, const uint8_t * restrict src, size_t max_count)
 {
     uint8_t *dstp = dst;
     const uint8_t *srcp = src;
 
     if (fi->type == NGLI_TYPE_MAT3) {
         const size_t dst_vec_stride = fi->stride / 3;
         const size_t src_vec_stride = sizes_map[NGLI_TYPE_VEC3];
-        const size_t count = 3 * NGLI_MAX(fi->count, 1);
+        const size_t count = 3 * NGLI_MAX(max_count ? max_count : fi->count, 1);
```

Comment:
To use the `count` argument name, you can rename the local `count` variable to `n`, so the function prototype is cleared.

Reply:
Added an intermediate rename commit and made this change.